Changed CMAKE_CXX_FLAGS to c++17, add target links to Abseil library and c_str() addition when logging

fordth opened the pull request:
*Issue #, if available:*

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

fordth committed:
Update CMakeLists.txt

CMakeLists link and path changes

fordth committed:
Update CMakeLists.txt

Added include paths for x86_64-linux-gnu and resolving for absl lib

fordth committed:
Update gmsa_service.cpp, added c_str() when logging variable values

fordth committed:
Update krb.cpp Added c_str() for logging

fordth committed:
Update build-using-csc.shchanged dotnet home path to /usr/lib/dotnet

fordth committed:
Update daemon.cpp

fordth committed:
Update renewal.cpp Added c_str() for logging

jamolina commented:
Thank you for your contribution! We will review your changes and get back to you as soon as possible

fordth committed:
Create Dockerfile-ubuntu-20.04

fordth committed:
Create Dockerfile-ubuntu-22.04

fordth committed:
Update Dockerfile-ubuntu-20.04 Refactored

fordth committed:
Update Dockerfile-ubuntu-22.04 Refactored

fordth committed:
Update Dockerfile-ubuntu-22.04

fordth committed:
Add git ignore

smhmhmd committed:
Conditional compilation for Ubuntu

* Changes in CMakeLists.txt with check for Ubuntu
* Detect location of .NET for Ubuntu

fordth committed:
Merge pull request #1 from smhmhmd/mainline

Conditional compilation for Ubuntu

smhmhmd committed:
Pull in grpc and krb5 compilation into credentials-fetcher

   If libraries are not available, compile grpc and krb5 libs and include
    into credentials-fetcher.
    grpc and krb5 libs are separate CMake targets.

The grpc and krb5 tar files are too large for github, they have to be downloaded.

To build grpc or krb5:
* "cd dependencies && mkdir build && cd build && make build_grpc && make build_krb5"

smhmhmd commented:
@fordth I have added one more PR on top of yours - https://github.com/fordth/credentials-fetcher/pull/2

fordth committed:
Merge pull request #2 from smhmhmd/mainline

Pull in grpc and krb5 compilation into credentials-fetcher

smhmhmd commented:
@saikiranakula-amzn - Please review PR

fordth commented:
@smhmhmd I just tried running cmake -DCMAKE_CXX_STANDARD=17 ../ on Ubuntu 20.04 and have now since received a number of errors. I will follow up with an output of the errors today ,but performing a cmake on the PR will repo the issue in Ubuntu

smhmhmd commented:
> @smhmhmd I just tried running cmake -DCMAKE_CXX_STANDARD=17 ../ on Ubuntu 20.04 and have now since received a number of errors. I will follow up with an output of the errors today ,but performing a cmake on the PR will repo the issue in Ubuntu

Hi @fordth ,
You dont need to run  cmake -DCMAKE_CXX_STANDARD=17 ../., cmake ../ is sufficient if you already have gRPC installed.
To install gRPC and krb5, you need to run cmake in the dependencies directory.
I have the steps in the comment of the PR - https://github.com/fordth/credentials-fetcher/pull/2

I just retested in my instance.

smhmhmd commented:
@fordth, I have some time till the end of the hour if you want to meet or we can meet on Monday.

smhmhmd commented:
@fordth
If you are able to test, I will get your PR merged.

fordth commented:
> 

We plan to test the compilation morning of Wednesday 9/13. Once this PR is then completed, I will create a new PR containing the additional changes we spoke about.  I will provide an update tomorrow

smhmhmd committed:
Set grpc,krb5 builds to install on default location

Changing grpc install to default location from user defined directory
since Cmake module in grpc searches only in default locations

This also includes docker changes to build on Ubuntu-20.04

smhmhmd committed:
Set grpc,krb5 builds to install on default location

Changing grpc install to default location from user defined directory
since Cmake module in grpc searches only in default locations

smhmhmd committed:
Merge branch 'mainline' of https://github.com/smhmhmd/credentials-fetcher-fordth into mainline

fordth committed:
Added -j4 to make instead of -j for parallelism

fordth committed:
Merge pull request #4 from fordth/pr

Contains docker file for Ubuntu 20.04 to support the latest CMake refactoring

ramapullareddyc commented on docker/Dockerfile-ubuntu-22.04:
```diff
@@ -0,0 +1,31 @@
+RUN apt-get update \
+    && apt-get install -y dotnet-sdk-6.0
+
+RUN cd /root && git clone https://github.com/fordth/credentials-fetcher \
```
Should we change the URL to point to correct branch?

fordth committed:
Update Dockerfile-ubuntu-20.04 to clone aws branch

fordth committed:
Update Dockerfile-ubuntu-22.04 to clone aws branch

fordth commented:
Good catch, I updated the PR

On Tue, Sep 19, 2023 at 1:46 PM Ram Chennuru ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In docker/Dockerfile-ubuntu-22.04
> <https://github.com/aws/credentials-fetcher/pull/64#discussion_r1330500819>
> :
>
> > +    && apt install -y git clang wget cmake curl \
> +        libglib2.0-dev libboost-dev libkrb5-dev libsystemd-dev libssl-dev \
> +        libboost-program-options-dev libboost-filesystem-dev
> +
> +RUN cd /root && git clone --recurse-submodules https://github.com/grpc/grpc \
> +   && cd grpc/ \
> +    && mkdir build \
> +    && cd build \
> +    && cmake -DgRPC_INSTALL=ON -DgRPC_BUILD_TESTS=OFF -DCMAKE_CXX_STANDARD=17 ../ \
> +    && make -j4 \
> +    && make install
> +
> +RUN apt-get update \
> +    && apt-get install -y dotnet-sdk-6.0
> +
> +RUN cd /root && git clone https://github.com/fordth/credentials-fetcher \
>
> Should we change the URL to point to correct branch?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/aws/credentials-fetcher/pull/64#pullrequestreview-1633878802>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACNAMAAQR65DUMMKTXRUXEDX3HLABANCNFSM6AAAAAA4GNCL2A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ramapullareddyc commented on docker/Dockerfile-ubuntu-20.04:
```diff
@@ -0,0 +1,39 @@
+FROM ubuntu:20.04 as base
+
+RUN apt-get update \
+    && DEBIAN_FRONTEND="noninteractive" TZ="America/New_York" \
```
UTC time zone is better and applicable universally and/or passing in a time zone would also be good. Same for 22.04 docker file as well.

smhmhmd replied:
Yes, an environment variable can be passed into Docker using --env option.

fordth replied:
The PR has been updated to require --build-arg TIME_ZONE during docker build

ramapullareddyc commented on docker/Dockerfile-ubuntu-20.04:
```diff
@@ -0,0 +1,39 @@
+    && apt-get install -y dotnet-sdk-6.0 \
+    && ln -s '/usr/share/dotnet' '/usr/lib/dotnet'
+
+RUN cd /root && git clone https://github.com/fordth/credentials-fetcher \
```
When do we plan to update URL to point to official repository? Same for 22.04 docker file as well.

fordth replied:
The PR has been updated with the correct repository

ramapullareddyc commented on docker/Dockerfile-ubuntu-22.04:
```diff
@@ -0,0 +1,31 @@
+FROM ubuntu:22.04 as base
+
+RUN DEBIAN_FRONTEND="noninteractive" TZ="America/New_York" \
```
Do we not need to build cMake and grpc like how we are doing in ubuntu-20.04 docker file?

smhmhmd replied:
The Dockerfile for Ubuntu-22.04 needs to be updated with changes from the Dockerfile for Ubuntu-20.04

fordth replied:
I can work on the ubuntu 22.04 Docker file but I may need a day or two to complete it.

ramapullareddyc commented on CMakeLists.txt:
```diff
@@ -126,15 +133,18 @@ add_custom_command(
         kadm5srv_mit kdb5 gssrpc gssapi_krb5 gssrpc k5crypto
         com_err krb5support resolv)
-
-target_link_libraries(credentials-fetcherd
```
Do we need to remove this line for non-ubuntu distros? "api/CMakeLists.txt" file links using same variables for non-ubuntu code path.

smhmhmd replied:
+1, kerberos libs are duplicated, this can be cleaned up.

fordth replied:
My latest update in the PR now includes the updated Ubuntu 22.04 docker file.

fordth committed:
Updated Docker file to support compiling krb5 and grpc as seperate downloads. Fixed issue with .net now included in Ubuntu 22.04 causing conflicts when installing .Net sdk 6 package.

ford-th committed:
Added required --build-arg TIME_ZONE

saikiranakula-amzn reviewed (approved):
testing on fedora
